lib/matplotlib/legend.py

```diff
@@ -286,6 +286,13 @@ def __init__(self, parent, handles, labels,
             ncol = 1
         self._ncol = ncol
 
+        if (
```

Comment:
It is probably better to do something like `if int(n) != n: raise`. 

Should probably make `numpoints` a property and do the validation on the way in as well.

This should probably also check that numpoints > 0.